docs/topics/scripting/custom-script-deps-tutorial.md

```diff
@@ -0,0 +1,330 @@
+   checkbox if you want to write the build script in Kotlin.
+
+   ![Create a root project for custom Kotlin scripting](script-deps-create-root-project.png){width=800}
+
+   Now you have an empty Kotlin/JVM Gradle project where you will add the required modules: script definition and scripting host.
+
+3. Go to **File | New | Module** and add a new **Gradle** module with **Kotlin/JVM**. Select the **Kotlin DSL build script**
+   checkbox if you want to write the build script in Kotlin. This module will be the script definition.
+
+4. Give the module a name and select the root module as its parent.
+
+   ![Create script definition module](script-deps-module-definition.png){width=800}
+
+5. In the module's `build.gradle(.kts)` file, remove the `version` of the Kotlin Gradle plugin. You already have it in the
+   root project's build script.
+
+6. Repeat steps 3, 4, and 5 one more time to create a module for the scripting host.
+
+The project should have the following structure:
+
+![Custom scripting project structure](script-deps-project-structure.png){width=300}
+
+You can find an example of such project and more Kotlin scripting examples in the [kotlin-script-examples GitHub repository](https://github.com/Kotlin/kotlin-script-examples/tree/master/jvm/basic/jvm-maven-deps).
+
+## Create a script definition
+
+1. In the script definition module, add the dependencies on the Kotlin scripting components in the `dependencies` block of
+   `build.gradle(.kts)`. These dependencies provide the APIs you will need for the script definition:
+
+   <tabs group="build-script">
+   <tab title="Kotlin" group-key="kotlin">
+
+   ```kotlin
+   dependencies {
+       implementation("org.jetbrains.kotlin:kotlin-scripting-common")
+       implementation("org.jetbrains.kotlin:kotlin-scripting-jvm")
+       implementation("org.jetbrains.kotlin:kotlin-scripting-dependencies")
+       implementation("org.jetbrains.kotlin:kotlin-scripting-dependencies-maven")
+       // coroutines dependency is required for this particular definition
+       implementation("org.jetbrains.kotlinx:kotlinx-coroutines-core:%coroutinesVersion%") 
+   }
+   ```
+
+   </tab>
+   <tab title="Groovy" group-key="groovy">
+
+   ```groovy
+   dependencies {
+       implementation 'org.jetbrains.kotlin:kotlin-scripting-common'
+       implementation 'org.jetbrains.kotlin:kotlin-scripting-jvm'
+       implementation 'org.jetbrains.kotlin:kotlin-scripting-dependencies'
+       implementation 'org.jetbrains.kotlin:kotlin-scripting-dependencies-maven'
+       // coroutines dependency is required for this particular definition
+       implementation 'org.jetbrains.kotlinx:kotlinx-coroutines-core-jvm:%coroutinesVersion%'
+
+   }
+   ```
+
+   </tab>
+   </tabs>
+
+5. Create the `src/main/kotlin/` directory in the module and add a Kotlin source file, for example, `scriptDef.kt`.
+
+6. In `scriptDef.kt`, create an abstract class. It will be the _script definition_.
```

Comment:
I'm not sure that we can write here "the _script definition_", it sounds like we're giving a definition here, and we're not yet sure about the terminology. Maybe it is better to write something like "It will serve as a reference to the script definition" or maybe "in addition to be used as a superclass for scripts of this type (hence `abstract`), it will serve as a reference to the script definition later".
It is also important to make it `abstract` or `open` - omitting it may cause some weird errors, so I'd put emphasis on it somewhere again, maybe in the comments.